error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/Slf4jLogDeclaration.java

```diff
@@ -0,0 +1,136 @@
+import com.sun.source.util.TreeScanner;
+import javax.inject.Inject;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+import javax.tools.JavaFileObject;
+import org.jspecify.annotations.Nullable;
+import tech.picnic.errorprone.bugpatterns.util.SourceCode;
+
+/**
+ * A {@link BugChecker} that warns when SLF4J declarations are not canonicalized across the project.
+ *
+ * @apiNote The default canonicalized logger name can be overriden through {@link ErrorProneFlags
+ *     flag arguments}.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Make sure SLF4J log declarations follow the SLF4J conventions inside classes.",
+    link = BUG_PATTERNS_BASE_URL + "Slf4jLogDeclaration",
+    linkType = CUSTOM,
+    severity = WARNING,
+    tags = LIKELY_ERROR)
+public final class Slf4jLogDeclaration extends BugChecker implements ClassTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<Tree> LOGGER = isSubtypeOf("org.slf4j.Logger");
+  private static final String CANONICALIZED_LOGGER_NAME_FLAG =
+      "Slf4jLogDeclaration:CanonicalizedLoggerName";
+  private static final String DEFAULT_CANONICALIZED_LOGGER_NAME = "LOG";
+  private static final Matcher<ExpressionTree> GET_LOGGER_METHOD =
+      staticMethod().onDescendantOf("org.slf4j.LoggerFactory").named("getLogger");
+
+  private final String canonicalizedLoggerName;
+
+  /** Instantiates a default {@link Slf4jLogDeclaration} instance. */
+  public Slf4jLogDeclaration() {
+    this(ErrorProneFlags.empty());
+  }
+
+  /**
+   * Instantiates a customized {@link Slf4jLogDeclaration}.
+   *
+   * @param flags Any provided command line flags.
+   */
+  @Inject
+  Slf4jLogDeclaration(ErrorProneFlags flags) {
+    canonicalizedLoggerName = getCanonicalizedLoggerName(flags);
+  }
+
+  @Override
+  public Description matchClass(ClassTree tree, VisitorState state) {
+    SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+
+    for (Tree member : tree.getMembers()) {
+      if (LOGGER.matches(member, state)) {
+        if (tree.getKind() != Kind.INTERFACE) {
+          fixLoggerVariableModifiers(member, state, fixBuilder);
+        }
+        canonicalizeLoggerVariable(member, state, fixBuilder);
+      }
+    }
+    fixLoggerVariableDeclaration(state, fixBuilder);
+
+    return fixBuilder.isEmpty() ? Description.NO_MATCH : describeMatch(tree, fixBuilder.build());
+  }
+
+  private static void fixLoggerVariableModifiers(
+      Tree member, VisitorState state, SuggestedFix.Builder fixBuilder) {
+    SuggestedFixes.addModifiers(member, state, Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+        .ifPresent(fixBuilder::merge);
+  }
+
+  private void canonicalizeLoggerVariable(
+      Tree member, VisitorState state, SuggestedFix.Builder fixBuilder) {
+    VariableTree variable = (VariableTree) member;
+    if (!variable.getName().toString().equals(canonicalizedLoggerName)) {
```

Comment:
```suggestion
    if (!variable.getName().contentEquals(canonicalizedLoggerName)) {
```
I think this would work too.

Reply:
Indeed; as I noticed this also [here](https://github.com/PicnicSupermarket/error-prone-support/pull/794/files#r1818105009), I proposed a Refaster rule for this: #1379.